ops/map_importer.py

```diff
@@ -130,16 +151,38 @@ def modal(self, context, event):
         if event.type == 'TIMER' and not self._updating:
             self._updating = True
 
-            for x in range(10):
-                try:
-                    self.import_object(context)
-                except:
-                    print("Can`t import model... skipping")
+            # Load collision files first if there are any left to load
+            if len(self._col_files) > 0:
+                filename = self._col_files.pop()
+                collection = bpy.data.collections.new(filename)
+                bpy.context.scene.collection.children.link(collection)
+                col_list = col_importer.import_col_file(os.path.join(self.settings.dff_folder, filename), filename)
+                # Move all collisions to a top collection named for the file they came from
+                for c in col_list:
+                    bpy.context.scene.collection.children.unlink(c)
+                    collection.children.link(c)
+                # Hide this collection in the viewport (individual collision meshes will be linked and transformed
+                # as needed to their respective map sections, this collection is just for export)
+                context.view_layer.active_layer_collection = context.view_layer.layer_collection.children[-1]
+                bpy.context.view_layer.active_layer_collection.hide_viewport = True
+
+                # Update cursor progress indicator if something needs to be loaded
+                num = (
+                        1.0 - float(len(self._col_files)) / float(self._col_files_to_load)
+                ) if self._col_files else 0
+
+            else:
+                for x in range(10):
```

Comment:
Are the objects being imported 10 times here? Is there any purpose for this or was this an oversight in the old code?

Reply:
That's just how it was in the old code. I believe it's loading 10 objects at a time here on each pulse of the timer. import_obj() increments the instance index itself, so nothing is being repeated it's just getting called 10x.

Reply:
Ah okay gotcha.